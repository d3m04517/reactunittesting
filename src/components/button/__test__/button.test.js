import React from 'react';
import ReactDom from 'react-dom';
import Button from '../button.component';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Button></Button>, div);
});

test("it renders button correctly", () => {
    const {getByTestId} = render(<Button label="I am a button."></Button>)
    expect(getByTestId('button')).toHaveTextContent("I am a button.");
});

test("matches snapshot", () => {
    const tree = renderer.create(<Button label="I am a button."></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});
test("matches snapshot 1", () => {
    const tree = renderer.create(<Button label="I am a button."></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});
test("matches snapshot 2", () => {
    const tree = renderer.create(<Button label="I am a button."></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});