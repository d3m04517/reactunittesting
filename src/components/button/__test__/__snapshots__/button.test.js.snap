// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="button-style"
  data-testid="button"
>
  I am a button.
</button>
`;

exports[`matches snapshot 1 1`] = `
<button
  className="button-style"
  data-testid="button"
>
  I am a button.
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="button-style"
  data-testid="button"
>
  I am a button.
</button>
`;
